cpp/src/io/orc/orc.hpp

```diff
@@ -528,20 +532,24 @@ class ProtobufWriter {
                            TypeKind kind,
                            ColStatsBlob const* stats);
 
+  void resize(std::size_t bytes) { m_buf->resize(bytes); }
+
+  std::unique_ptr<std::vector<uint8_t>> release() { return std::move(m_buf); }
+
  public:
-  size_t write(const PostScript&);
-  size_t write(const FileFooter&);
-  size_t write(const StripeInformation&);
-  size_t write(const SchemaType&);
-  size_t write(const UserMetadataItem&);
-  size_t write(const StripeFooter&);
-  size_t write(const Stream&);
-  size_t write(const ColumnEncoding&);
-  size_t write(const StripeStatistics&);
-  size_t write(const Metadata&);
+  size_t write(PostScript const&);
+  size_t write(FileFooter const&);
+  size_t write(StripeInformation const&);
+  size_t write(SchemaType const&);
+  size_t write(UserMetadataItem const&);
+  size_t write(StripeFooter const&);
+  size_t write(Stream const&);
+  size_t write(ColumnEncoding const&);
+  size_t write(StripeStatistics const&);
+  size_t write(Metadata const&);
 
  protected:
-  std::vector<uint8_t>* m_buf;
+  std::unique_ptr<std::vector<uint8_t>> m_buf;
```

Comment:
why not just std::vector?

Reply:
Because with `unique_ptr`, we can make sure any `put_` call after `release()` will throw an exception. Otherwise, we may accidentally put new data into the buffer.

Reply:
now that release has been removed, can we remove unique_ptr?

Reply:
Yes, the reason to have `unique_ptr` is not as compelling as before. I'll remove it.